frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/capacity-and-nodes.tsx

```diff
@@ -49,3 +75,102 @@ type SelectNodesDetailsProps = {
   zones: number;
   memory: number;
 };
+
+export const StretchClusterFormGroup: React.FC<stretchClusterFormGroupProps> = ({
+  state,
+  dispatch,
+  pvData,
+  nodesData,
+}) => {
+  const { t } = useTranslation();
+  const { stretchClusterChecked, selectedArbiterZone, storageClass, nodes } = state;
+  const [zonesOption, setZonesOptions] = React.useState({});
+  const [zones, setZones] = React.useState([]);
+  const [showInfoAlert, setShowInfoAlert] = React.useState(true);
+  const [storageClassData] = useK8sWatchResource<StorageClassResourceKind[]>(scResource);
+
+  const nodesDataMemoized: NodeKind[] = useDeepCompareMemoize(nodesData, true);
+
+  const isArbiterDisabled = React.useCallback(
+    (): boolean => !storageClassData?.some((sc) => isArbiterSC(sc, pvData, nodesDataMemoized)),
+    [nodesDataMemoized, pvData, storageClassData],
+  );
+
+  React.useEffect(() => {
+    if (!stretchClusterChecked && isArbiterDisabled()) {
+      dispatch({ type: 'setStretchClusterChecked', value: false });
+    }
+    if (stretchClusterChecked) {
+      const uniqZones: Set<string> = new Set(nodesDataMemoized.map((node) => getZone(node)));
+      const uniqSelectedNodesZones: Set<string> = new Set(nodes.map((node) => getZone(node)));
+      setZones(_.difference([...uniqZones], [...uniqSelectedNodesZones]));
+    }
+  }, [storageClass, stretchClusterChecked, nodes, isArbiterDisabled, dispatch, nodesDataMemoized]);
+
+  React.useEffect(() => {
+    if (stretchClusterChecked) {
+      setZonesOptions(_.zipObject(zones, zones));
+      dispatch({ type: 'setSelectedArbiterZone', value: zones[0] });
+    }
+  }, [dispatch, stretchClusterChecked, zones]);
+
+  return (
+    <FormGroup fieldId="arbiter-cluster" label={t('ceph-storage-plugin~Stretch Cluster')}>
+      <Checkbox
+        aria-label="Enable Arbiter"
+        id="arbiter-cluster"
+        isChecked={stretchClusterChecked}
+        label="Enable arbiter"
+        description={t(
+          'ceph-storage-plugin~To support high availability when two data centers can be used, enable arbiter to get the valid quorum between two data centers.',
+        )}
+        isDisabled={isArbiterDisabled()}
+        onChange={(isChecked: boolean) =>
+          dispatch({ type: 'setStretchClusterChecked', value: isChecked })
+        }
+      />
+      {showInfoAlert && (
+        <Alert
+          aria-label="Arbiter minimum requirements"
```

Comment:
externalize this as well
```suggestion
          aria-label="Arbiter minimum requirements"
```